source/sliding_puzzle.py

```diff
@@ -1,3 +1,5 @@
+import random
```

Comment:
```suggestion
from random import choice
```

See below

Reply:
Good catch, I'll fix it

Reply:
for this I will keep the ```import random``` statement,  I'm using the shuffle from the random module as well, the only reason to replace it with your suggestion would be optimization, but for a code base of this size, it is not that important